src/NzbDrone.Core/Datastore/Migration/008_redacted_api.cs

```diff
@@ -0,0 +1,21 @@
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(8)]
+    public class redacted_api : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            // Remove username/password fields for Redacted
+            Execute.Sql("UPDATE Indexers Set Settings = JSON_REMOVE(Indexers.Settings, '$.username', '$.password') WHERE Implementation = 'Redacted'");
```

Comment:
One thing to consider is that these JSON functions were not added to sqlite until 3.9 or so. This means default package on Centos7, etc.. may fail this migration. Even though 3.9 is ancient, I'm hesitant to use this in a migration given the issues we saw on Radarr with this.

Reply:
Ok, makes sense.  I've switched it over to deserializing the settings json.